src/exec/builtins/functions.rs

```diff
@@ -134,42 +149,85 @@ pub fn list_directory(shell: &mut ShellState, args: Vec<&str>) -> Result<()> {
 
     directories.append(&mut files);
 
-    let permission_view = match octal_permissions {
-        true => DirectoryListPermissionMode::Octal,
-        false => DirectoryListPermissionMode::String,
+    let permission_view = { 
+        if hide_permissions {
+            DirectoryListPermissionMode::Hidden
+        } else {
+            match octal_permissions {
+                true => DirectoryListPermissionMode::Octal,
+                false => DirectoryListPermissionMode::String,
+            }
+        }
     };
 
-    list_directory_long(directories, permission_view, path_to_read.clone(), permission_seperator);
+    let timestamp =  {
+        if hide_timestamps {
+            DirectoryListTimestampMode::Hidden
+        } else {
+            let mut seen_true = 0;
+            let mut true_indx = 0;
+            let times = [
+                use_modified_time,
+                use_accessed_time,
+                use_created_time
+            ];
+
+            for (i, timemode) in times.iter().enumerate() {
+                if *timemode {
+                    seen_true += 1;
+                    true_indx = i;
+                }
+            };
+
+            match seen_true {
+                1 => {
+                    match true_indx {
+                        0 => DirectoryListTimestampMode::Modified,
+                        1 => DirectoryListTimestampMode::Accessed,
+                        2 => DirectoryListTimestampMode::Created,
+                        _ => DirectoryListTimestampMode::Modified,
+                    }
+                },
+                _ => DirectoryListTimestampMode::Modified,
+            }
+        }
+    };
+
+    list_directory_long(directories, permission_view, path_to_read.clone(), permission_seperator, timestamp, hide_user, hide_file_sizes);
 
     Ok(())
 }
 
-fn list_directory_long(item: Vec<String>, permission_format: DirectoryListPermissionMode, cwd: PathBuf, permission_seperator: bool) {
+fn list_directory_long(item: Vec<String>, permission_format: DirectoryListPermissionMode, cwd: PathBuf, permission_seperator: bool, timestamp: DirectoryListTimestampMode, hide_user: bool, hide_file_sizes: bool) {
     let path_to_read = cwd;
     let mut file_size_len: usize = 0;
     let mut file_size_len_last: usize = 0;
```

Comment:
Please be running Clippy in your editor and also setting it to autoformat with Rustfmt.